Initial design changes

jelliotartz opened the pull request:
## Describe Changes
- Towards #122.
- Makes text input labels uniform in style - uses "Name" on Edit Details page as desired style.
- Text size increased:
  - Help text size beneath text inputs increased
  - Removed `font-size` on `form_label` to increase font size
- Delete button size reduced.
- Adds more space added divider lines.
- Space under text input labels is consistent.
  - RFC: use of custom components, eg `DisplayNameSlugEditor` and `MarkdownEditor`, along with HTML inputs made it difficult for me to elegantly target the elements - current solution (below) attempts to stay within BEM conventions, but suggestions on alternative approaches are welcome.
  ```javascript
    &__label
    input.slug-editor__form-input
      margin-top: 1em
    &--top-margin
      margin-top: 1em
    ```
- About page text input label has same style as Edit Details page.
- Adds `Create project` button on org/projects page.
- [Staged here](https://edit-details-page-styling.lab-preview.zooniverse.org/).

Not labeling this as a WIP PR, but understand that this is iterative, and there may be more elegant/simple/legible solutions, or better practices, that accomplish the desired results.

## Known Issues
- The MarkdownEditor WYSIWYG buttons have lost their styling. TODO: discuss desired design.

eatyourgreens commented:
I had a quick look at the `MarkdownEditor` code, and it takes a `className` prop:
https://github.com/zooniverse/markdownz/blob/master/src/components/markdown-editor.jsx#L143
So, you could add your CSS classes directly on the component itself `<MarkdownEditor className=…` and they'll be added to the element, in addition to the component's own CSS classes.

If `DisplayNameSlugEditor` doesn't take a `className` prop already, I'd suggest updating it so that it does.

eatyourgreens commented on src/modules/organizations/containers/details-form-container.jsx:
```diff
@@ -68,12 +68,14 @@ class DetailsFormContainer extends React.Component {
-              resourceType="organization"
-              ref={(node) => { this.fields.display_name = node; }}
-            />
+            <label className="form__label" htmlFor="display_name">
```
`htmlFor` isn't necessary when the label wraps the input. It's used to link a label and input when they aren't implicitly connected in the markup.

The a11y linter might warn about it being missing, but you can ignore than warning in cases like this.

eatyourgreens commented on src/modules/organizations/containers/details-form-container.jsx:
```diff
@@ -93,14 +95,16 @@ class DetailsFormContainer extends React.Component {
-              />
+              <div className="form__input--top-margin">
+                <MarkdownEditor
+                  id="introduction"
```
You can add `className="form__input--top-margin"` here, which avoids the awakward wrapping `div`.

eatyourgreens commented on src/modules/organizations/containers/details-form-container.jsx:
```diff
@@ -93,14 +95,16 @@ class DetailsFormContainer extends React.Component {
-                rows="10"
-                value={this.state.textarea}
-              />
+              <div className="form__input--top-margin">
```
Not a big problem, but I don't think `div` is a valid child of `label` in HTML. If you need a generic wrapper inside a label, use `span`.

If this div is just here in order to apply the CSS class to the Markdown editor, you can add that directly to the `MarkdownEditor` component.

eatyourgreens commented on src/modules/organizations/containers/links-container.jsx:
```diff
@@ -45,31 +45,35 @@ class LinksContainer extends React.Component {
-            />
-          </span>
+            <div className="form__input--top-margin">
+              <ExternalLinksEditor
```
We should add a className` prop to this component, if it doesn't have one already, to avoid the wrapper `div`.

eatyourgreens commented on src/modules/organizations/containers/links-container.jsx:
```diff
@@ -45,31 +45,35 @@ class LinksContainer extends React.Component {
-          </span>
+            <div className="form__input--top-margin">
+              <SocialLinksEditor
+                id="social"
```
We should add a `className` prop to this component, if it doesn't have one already, to avoid the wrapper `div`.

jelliotartz commented:
## PR Update
Thanks for your suggestions @eatyourgreens! 
- `DisplayNameSlugEditor` does indeed have a `className` prop. I refactored the css selector a little, to target the default class name for its input `.slug-editor__form-input`. Conferred with Mark and we decided not to override the default class name of slug-editor (at least for now).
- I added `form__input--top-margin` on the `className` prop for `ExternalLinksEditor` and `SocialLinksEditor`.
- Conferred with @beckyrother about `line-height` attribute for ImageSelector help text and `margin-top` value.

srallen commented:
If the design for `DisplayNameSlugEditor` is changing, then those style defaults can be modified or removed as needed. Its className prop could also not be set to that default class and be independently set and not a override. ZRC shouldn't be thought of as a unchangeable monolith. It hasn't even gotten its version 1 release yet because I left it to be easily modified as initial development is happening for its use.

srallen commented:
Just so its known, the only other place that `DisplayNameSlugEditor` is used is with collections and that's why it is in the ZRC repo. The collections editor doesn't really have a design, so I wouldn't worry about trying to work this with that.

jelliotartz commented:
## PR Update
Thanks for your input, @srallen! 

- Cleaned up a few details missed in last commit (extraneous htmlFor on label tag, and wrapping div on About page's MarkdownEditor.
- Added independent class for DisplayNameSlugEditor, rather than overriding the default class. Applied some of default styles, but adjusted `margin-top` on input and `font-size` on help text.

mcbouslog commented on src/styles/global/form.styl:
```diff
@@ -9,15 +9,30 @@
-    font-weight: bold
-    letter-spacing: .1em
-    opacity: .7
+    .org-details-slug-editor__form-input
```
it feels strange to me this and `.org-details-slug-editor__form-help` is here in the `global/form.styl`, do these belong in `components/edit-details.styl`? maybe check where else `.form__label` used and leave behind some basic styling.

jelliotartz commented:
## PR Update
- `.org-details-slug-editor__form-input` and `__help` moved to `edit-details.styl`.
- Let me know if you'd like any other styling applied to `.form__label` :)

beckyrother commented:
This looks great @jelliotartz ! 
One small thing: can you center the "Drop an image here" text both vertically, horizontally, and text-align? 
![zoo-lab](https://user-images.githubusercontent.com/1392130/32393851-2790d47a-c0a9-11e7-8dac-45adc50df357.png)

srallen commented on src/modules/common/components/social-links-editor.jsx:
```diff
@@ -78,7 +78,7 @@ export default class SocialLinksEditor extends React.Component {
     return (
       <div>
-        <table className="edit-social-links">
+        <table className="edit-social-links form__input--top-margin">
```
I really don't think a class intended for form defaults should be reused on tables even if it's just for applying a margin. This sets us up for unintended collisions which is an issues with PFE's current CSS.

beckyrother commented:
visually lgtm!

mcbouslog commented:
spoke with @beckyrother and @jelliotartz - we're going to table the label or heading margin for now, think about how we want to approach it (possibly edit ZRC DisplayNameSlugEditor component) and then make any necessary changes.
